File: mad3004-introd-swift/C0711561_F2017_MAD3004_FinalProject/C0711561_F2017_MAD3004_FinalProject/FixedBasedPartTime.swift

public class FixedBasedPartTime: PartTime {
    
    private var fixedAmount: Double
    
    override init() {
        fixedAmount = 0.0
        super.init()
    }
    
    init(ppName: String, ppAge: Int, pHourlyRate: Int, pNumberHoursWorked: Int, fbFixedBasedPartTime: Double) {
        self.fixedAmount = fbFixedBasedPartTime
        super.init(ppName: ppName, ppAge: ppAge, pHourlyRate: pHourlyRate, pNumberHoursWorked: pNumberHoursWorked)
    }
    
    init(ppName: String, ppAge: Int, pHourlyRate: Int, pNumberHoursWorked: Int, ppV: Vehicle, fbFixedBasedPartTime: Double) {
        self.fixedAmount = fbFixedBasedPartTime
        super.init(ppName: ppName, ppAge: ppAge, pHourlyRate: pHourlyRate, pNumberHoursWorked: pNumberHoursWorked, ppV: ppV)
    }
    
    private func calcSalary() -> Double {
        return Double(hourlyRate * numberHoursWorked)
    }
    
    override func calcEarnings() -> Double {
        return calcSalary() + fixedAmount
    }
    
    override func getTypeOfEmployee() -> String {
        return super.getTypeOfEmployee() + " / Fixed Amt"
    }
    
    override func printMyData() -> String {
        return "\(super.printMyData())\n  -Fixed Amount: \(Util.getInstant().toCurrencyFormatFrom(value: fixedAmount))\n  -Earnings: \(Util.getInstant().toCurrencyFormatFrom(value: calcEarnings())) (\(calcSalary()) + \(fixedAmount))"
    }
    
}

File: mad3464-programming-java-i/C0711561_F2017_MAD3464_FinalProject/src/c0711561/f2017/mad3464/finalproject/employee/FixedBasedPartTime.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package c0711561.f2017.mad3464.finalproject.employee;

import c0711561.f2017.mad3464.finalproject.Util;
import c0711561.f2017.mad3464.finalproject.vehicle.Vehicle;
import java.math.BigDecimal;

/**
 *
 * @author macstudent
 */
public class FixedBasedPartTime extends PartTime {
    
    private double fixedAmount;

    public FixedBasedPartTime(String name, int age, double rate, int hoursWorked, double fixedAmount) {
        super(name, age, rate, hoursWorked);
        this.fixedAmount = fixedAmount;
    }
    
    public FixedBasedPartTime(String name, int age, double rate, int hoursWorked, double fixedAmount, Vehicle vehicle) {
        super(name, age, rate, hoursWorked, vehicle);
        this.fixedAmount = fixedAmount;
    }

    private double calcSalary() {
        return getRate() * getHoursWorked();
    }

    @Override
    public String getTypeOfEmployee() {
        return super.getTypeOfEmployee() + " / Fixed Amt";
    }
    
    @Override
    public double calcEarnings() {
        return fixedAmount + calcSalary();
    }

    @Override
    public String printMyData() {
        return super.printMyData() 
                + "\n - Fixed Amount: " + Util.toCurrencyFormatFrom(fixedAmount)
                + "\n - Earnings: " + Util.toCurrencyFormatFrom(calcEarnings()) + "(" + calcSalary() + " + " + fixedAmount + ")";
    }
}
